python/pyspark/sql/column.py

```diff
@@ -250,11 +250,50 @@ def __iter__(self):
         raise TypeError("Column is not iterable")
 
     # string methods
+    _rlike_doc = """
+    Return a Boolean :class:`Column` based on a regex match.
+
+    :param other: an extended regex expression
+
+    >>> df.filter(df.name.rlike('ice$')).collect()
+    [Row(name=u'Alice', age=1)]
```

Comment:
It sounds the test failure is here. If you click the link, the full logs can be checked (e.g., https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/75622/console).

Up to my knowledge, Row in python sorts the field names and therefore I guess it should be `[Row(age=1, name=u'Alice')]`.